tf_model/train_classifier.py

```diff
@@ -0,0 +1,265 @@
+        vectorizer = self.create_vectorizer(data)
+        emb_matrix = self.create_emb_matrix(vectorizer)
+        
+        tweet_input = Input(shape=(1,), dtype=tf.string, name='tweet')
+        
+        x = vectorizer(tweet_input)
+        
+        x = Embedding(input_dim=self.vocab_size,
+                      output_dim=self.embed_dim,
+                      input_length=self.max_length,
+                      weights=[emb_matrix],
+                      trainable=False,
+                      name='GloVe_200')(x)
+        
+        x_1 = self.bigram(x)
+        x_1 = self.pool(x_1)
+
+        x_2 = self.trigram(x)
+        x_2 = self.pool(x_2)
+
+        x_3 = self.fourgram(x)
+        x_3 = self.pool(x_3)
+
+        x_4 = self.fivegram(x)
+        x_4 = self.pool(x_4)
+        
+        merged = self.concat([x_1, x_2, x_3, x_4])
+        
+        merged = self.dense(merged)
+        
+        if training:
+            merged = self.dropout(merged)
+
+        predictions = self.preds(merged)
+        
+        model = Model(tweet_input, predictions)
+
+        return model
+
+
+    def compile_model(self, model):
+
+        if self.num_classes == 2:
+            model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
+    
+        else:
+            model.compile(
+            loss='sparse_categorical_crossentropy', optimizer='adam', metrics=['sparse_categorical_accuracy'])
+    
+        return model
+
+
+    def custom_callbacks(self):
+
+        es = EarlyStopping(
+            monitor='val_sparse_categorical_accuracy',
+            mode='max',
+            patience=7,
+            verbose=1
+        )
+
+        plateau = ReduceLROnPlateau(
+            monitor='val_sparse_categorical_accuracy',
+            mode='max',
+            factor=0.1,
+            patience=3,
+            verbose=1
+        )
+
+        return es, plateau
+    
+    
+    def train(self):
+        all_tweets, all_labels = self.get_data()
+        train_data, train_labels, val_data, val_labels, _, _ = self.data_splits(all_tweets, all_labels)
+        es, plateau = self.custom_callbacks()
+        
+        model = self.call(train_data, training=True)
+
+        model = self.compile_model(model)
+        
+        model.fit(train_data,
+                  train_labels,
+                  batch_size=self.batch_size,
+                  epochs=self.epochs,
+                  validation_data=(val_data, val_labels),
+                  class_weight=self.get_weights(train_labels),
+                  callbacks=[es, plateau]
+                  )
+
+        return model
+    
+    def wrap_model(self):
+        model = self.train()
+        self.wrapper = CMPredict(model)
+        tf.saved_model.save(
+            self.wrapper.model, f'covid_mask_classifier/saved_models/{int(time())}',
+            signatures={'serving_default': self.wrapper.prediction}
+        )
+
+
+
+if __name__ == '__main__':
+    model = CMModel()
```

Comment:
the configurations you defined in `assets` are not used here, right? you're just using the default values?